courses/views.py

```diff
@@ -339,18 +339,50 @@ def dispatch(self, *args, **kwargs):
 
 @login_required
 def list_of_courses_to_add(request):
-    qs = Course.objects.all()
-    student = get_object_or_404(Student, name=request.user)
-    return render(
-        request,
-        'course_list_to_add.html',
-        {
-            'courses': qs,
-            'student_id': student.pk
-        }
-    )
+    if not request.user.is_instructor():
+        qs = Course.objects.all()
+        student = get_object_or_404(Student, name=request.user)
+        return render(
+            request,
+            'course_list_to_add.html',
+            {
+                'courses': qs,
+                'student_id': student.pk
+            }
+        )
+    else:
+        qs = Course.objects.all()
+        return render(
+            request,
+            'instructor_course_list.html',
+            {
+                'courses': qs,
+            }
+        )
 
 
+@login_required
+def my_list_of_courses(request):
+    if not request.user.is_instructor():
+        qs = Course.objects.filter(students=request.user.student)
+        student = get_object_or_404(Student, name=request.user)
+        return render(
+            request,
+            'student_course_list.html',
+            {
+                'courses': qs,
+                'student_id': student.pk
```

Comment:
u don't need the student_id variable

replace it where it is used in the template with
`request.user.pk`